.github/workflows/rust-audit.yml

```
@@ -0,0 +1,28 @@
+name: Rust security audit
+
+on:
+  schedule:
+    - cron: '0 2 * * *' # run at 2 AM UTC
+  push:
+    paths:
+      - '**/Cargo.toml'
+      - 'Cargo.lock'
+
+jobs:
+  security-audit:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'ci skip')"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/audit-check@v1.2.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Discord notification
+        if: failure()
```

```suggestion
        if: failure() && (github.event_name == 'push' && github.event.ref == 'refs/heads/master' || github.event_name == 'schedule')
```

We don't expect failures from PRs or non-master branches to be reported to Discord, so I'd suggest skipping this in these cases.

---

Seems smart, I can give it a go as soon as I'm able.